astropy/coordinates/baseframe.py

```diff
@@ -1248,6 +1248,41 @@ def is_frame_attr_default(self, attrnm):
         """
         return attrnm in self._attr_names_with_defaults
 
+    @staticmethod
+    def _frameattr_equiv(left_fattr, right_fattr):
+        """
+        Determine if two frame attributes are equival.  Implemented as a
+        staticmethod mainly as a convenient location, althought conceivable it
+        might be desirable for subclasses to override this behavior.
+
+        Primary purpose is to check for equality of representations, since by
+        default representation equality is only "is it the same object", which
+        is too strict for frame comparisons.
+
+        Note: this method may be removed when/if representations have an
+        appropriate equality defined.
+        """
+        if isinstance(left_fattr, r.BaseRepresentationOrDifferential):
+            if left_fattr is right_fattr:
+                # shortcut if it's exactly the same object
+                return True
+            elif isinstance(right_fattr, r.BaseRepresentationOrDifferential):
+                # both are representations.  Note that this ignores any
+                # stored differentials
```

Comment:
What about raising a warning if either ```right_fattr``` or ```left_fattr``` has differentials present?

Reply:
I'd go further and just return `False` - right now, we're even stricter and we can relax in steps.

Reply:
will do